cpp/src/aztec3/circuits/kernel/private/.test.cpp

```diff
@@ -70,288 +61,256 @@ using aztec3::circuits::apps::test_apps::escrow::deposit;
 
 using DummyComposer = aztec3::utils::DummyComposer;
 
-// using aztec3::circuits::mock::mock_circuit;
 using aztec3::circuits::mock::mock_kernel_circuit;
 
+// A type representing any private circuit function
+// (for now it works for deposit and constructor)
+using private_function = std::function<OptionalPrivateCircuitPublicInputs<NT>(
```

Comment:
I was going to ask why the args to the app circuits' function signatures were changed (in other files), and I guess this abstraction is the reason?

Reply:
That's right :thinking: trying to think of a better way. Do you think I overdid the generalization of this test helper logic?